### PR TITLE
Xeno Directional Attack IFF pref

### PR DESCRIPTION
**Pull request by PhantomEpicness**
## About The Pull Request

Added a preference to toggle a check that disables a friendly xeno from being caught in a directional attack. This is enabled by default.

It checks for xenos of the same hivenumber so XvX is unaffected

This is enabled by default.

it's also a verb that can be binded to a macro as needed with 
`directional_attack_iff_toggle`

## Why It's Good For The Game

massive qol + no more accidentally knocking down friendly xenos when trying to cap

## Changelog

:cl: totalepicness
qol: Added a preference to toggle a check that disables a friendly xeno from being caught in a directional attack. This is enabled by default.
qol: The above can be toggled by a verb if needed with "directional_attack_iff_toggle"
/:cl:

### CODE REVIEW

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -48,7 +48,7 @@ var/const/MAX_SAVE_SLOTS = 10
 	var/lastchangelog = ""				// Saved changlog filesize to detect if there was a change
 	var/ooccolor
 	var/be_special = 0				// Special role selection
-	var/toggle_prefs = TOGGLE_MIDDLE_MOUSE_CLICK|TOGGLE_DIRECTIONAL_ATTACK|TOGGLE_MEMBER_PUBLIC // flags in #define/mode.dm
+	var/toggle_prefs = TOGGLE_MIDDLE_MOUSE_CLICK|TOGGLE_DIRECTIONAL_ATTACK|TOGGLE_MEMBER_PUBLIC|TOGGLE_DIR_ASSIST_IFF // flags in #define/mode.dm
```
Comment (stanalbatross) [inline code/modules/client/preferences.dm:51]: if you are making it on by default you should update savefiles

File: code/_onclick/xeno.dm [hunk 2]
```diff
@@ -18,7 +18,10 @@
 				if (!alt)
 					alt = L // last option is a simple mob
 				continue
-
+			if (istype(L, /mob/living/carbon/Xenomorph) && client?.prefs?.toggle_prefs & TOGGLE_DIR_ASSIST_IFF)
+				var/mob/living/carbon/Xenomorph/X = L
+				if (X.hivenumber == src.hivenumber)
```
Comment (stanalbatross) [inline code/_onclick/xeno.dm:23]: make this a can_not_harm check

File: code/_onclick/xeno.dm [hunk 3]
```diff
@@ -18,7 +18,10 @@
 				if (!alt)
 					alt = L // last option is a simple mob
 				continue
-
+			if (istype(L, /mob/living/carbon/Xenomorph) && client?.prefs?.toggle_prefs & TOGGLE_DIR_ASSIST_IFF)
+				var/mob/living/carbon/Xenomorph/X = L
```
Comment (stanalbatross) [inline code/_onclick/xeno.dm:22]: better var name needed for `X`

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Comment by PhantomEpicness** [comment]
racist bot

**Pull request closed without merging**